enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+
+Phase 1 is pretty straightforward - simply adding an annotation to the
+existing manifest files in the various repositories for components to
+be included in single node developer clusters.
+
+Phase 2 is more of an open ended process leveraging the new profile to
+optimize the configuration of the individual components for single
+node developer clusters. In general, each component will be addressed
+independently and possibly multiple times, first to eliminate the
+post-install patching by SNC, and then to enable additional
+optimizations. During implementation of this phase, it is expected
+that patterns will emerge when common approaches apply across
+different components.
+
+### Risks and Mitigations
+
+One risk is that all parts of the implementation of the cluster
+profiles
+[enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+do not get merged in a timely manner. Implementing the
+single-node-developer cluster profile requires the ability to specify
+a non-default cluster profile to the OpenShift installer, and for this
+to result in only the manifests annotated for the specified cluster
+profile to be processed by the resulting cluster. Developer tools team
+members have proposed CVO patches as part of that enhancement, and
+will continue to participate in the review process, but progress
+getting the implementation merged has been slow so far. Having this
+enhancement as an additional use case for the cluster profile
+mechanism hopefully will help clarify how that feature is being used
+and expedite its implementation.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that should be included in single node
+developer clusters, but will not have the necessary annotation
+included in their manifests due to lack of awareness or concern for
+the CRC product. Given that single node developer clusters are
+expected to pass the majority of the OpenShift conformance tests,
+component developers should include this profile's annotation unless
+the new component is clearly not relevant to the development of
+applications on single node clusters. Developer tools team members
+will be available to collaborate with component developers regarding
+whether a component should be included in the single-node-developer
+cluster profile, and if so, how it might be configured to minimize
+resource requirements.
```

Comment:
Regarding that final risk, at the moment there is no way to differentiate between "yes, this component is excluded from profile XXX on purpose" and "developers of the component forgot to consider profile XXX", it would be nice to have a way to indicate the latter so that we can automatically detect missing profile information.

Reply:
Would it make sense to add an annotation such as:

include.release.openshift.io/single-node-developer=false

to a manifest to formally record the fact that this particular manifest is intentionally not included in the profile?

Reply:
Yeah, this probably will be useful to have once we start checking manifests for missing annotations (see https://github.com/openshift/oc/pull/618 on that  matter)